src/gtk/toplevel.cpp

```diff
@@ -1523,6 +1543,52 @@ wxTopLevelWindowGTK::DecorSize& wxTopLevelWindowGTK::GetCachedDecorSize()
     return size[index];
 }
 
+void wxTopLevelWindowGTK::GTKDoAfterShow()
+{
+    wxShowEvent showEvent(GetId(), true);
+    showEvent.SetEventObject(this);
+    HandleWindowEvent(showEvent);
+
+#ifdef __WXGTK3__
+    // Set the client size again if necessary, we should be able to do it
+    // correctly by now as the style cache should be up to date.
+    GTKUpdateClientSizeIfNecessary();
```

Comment:
It might be better to do this before generating the `wxShowEvent`.

Reply:
Good point, thanks, changed.